src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -195,9 +216,12 @@ private[health] class HealthCheckActor(
             health.update(result)
         })
     }
-    updatedHealth.onComplete {
-      case Success(newHealth) => self ! InstanceHealth(result, health, newHealth)
-      case Failure(t) => logger.error(s"An error has occurred: ${t.getMessage}", t)
+    Try(Await.ready(updatedHealth, 5 seconds)) match {
+      case Success(future) => future.value.get match {
+        case Success(newHealth) => self ! InstanceHealth(result, health, newHealth)
```

Comment:
With new implementation, if code timeout, then we might skip this line. two potential solutions : 
-  Make it fully blocking without timeout 
- Send a default value if it timeout